### PR TITLE
Integrate php cs fixer

### PR DESCRIPTION
**Pull request by PierreRambaud**
First step to centralize our coding standards

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "prestashop/php-coding-standards",
+  "description": "PrestaShop coding standard",
+  "version": "0.0.1",
+  "require": {
+    "symfony/console": "~3.2 || ~4.0",
+    "friendsofphp/php-cs-fixer": "^2.14",
+    "phpstan/phpstan": "^0.11.5",
```
Comment (Quetzacoalt91) [inline composer.json]: Well, that's what I was talking about.

In my opinion PHPStan has nothing to do here.

Comment (PierreRambaud) [reply composer.json]: It'll help to configure the boostrap.php & phpstan.neon which are specific to a PrestaShop configuration

**Comment by mickaelandrieu** [comment]
is it an issue if PHPStan require PHP 7? this means that you can't include this package in the core and all the core module right now

**Comment by PierreRambaud** [comment]
@mickaelandrieu Thinking about removing the phpstan package but keeping the boostrap & phpstan.neon files, wdyt?

**Comment by mickaelandrieu** [comment]
@PierreRambaud I think it's a good idea (and you can rely on PHPStan GitHub action if needed)

**Comment by PierreRambaud** [comment]
So it's ok now :)

File: templates/phpstan/phpstan.neon [hunk 2]
```diff
@@ -0,0 +1,11 @@
+parameters:
+  reportUnmatchedIgnoredErrors: false
+  bootstrap: /web/module/tests/php/phpstan/bootstrap.php
+  paths:
+    - /web/module/src
+  ignoreErrors:
+    # module specific
+    - '~Iterating over an object of an unknown class mysqli_result\.~'
+    - '~Access to offset mixed on an unknown class mysqli_result\.~'
```
Comment (Quetzacoalt91) [inline templates/phpstan/phpstan.neon]: These two lines should be removed

File: .php_cs.dist [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+$config = new PrestaShop\CodingStandards\CsFixer\Config();
+
+$config
+    ->setUsingCache(true)
+    ->getFinder()
+    ->in(__DIR__)
+    ->exclude('vendor');
+
+return $config;
```
Comment (Quetzacoalt91) [inline .php_cs.dist]: What is this file for? Checking the code of this repository? We should enable PrettyCI

Comment (PierreRambaud) [reply .php_cs.dist]: Yep it check this repository.
Agree to enable PrettyCI!

File: .php_cs.dist [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+$config = new PrestaShop\CodingStandards\CsFixer\Config();
```
Comment (Quetzacoalt91) [inline .php_cs.dist]:
```suggestion
$config = new PhpCsFixer\Config();
```

File: templates/cs-fixer/php_cs.dist [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?php
+
```
Comment (Quetzacoalt91) [inline templates/cs-fixer/php_cs.dist]: Something to require here?

Comment (PierreRambaud) [reply templates/cs-fixer/php_cs.dist]: Nope, this file is included by cs fixer

File: .php_cs.dist (hunk 4 above)
Comment (PierreRambaud) [reply .php_cs.dist]: Nope, we need to include "our" Cs Fixer configuration.

Comment (Quetzacoalt91) [reply .php_cs.dist]: Then you have to fix the require for PrettyCI. :)

**Pull request merged**